assets/src/edit-story/components/colorPicker/editableHexPreview.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { EditableInput } from 'react-color/lib/components/common';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useLayoutEffect, useState } from '@wordpress/element';
+
+const Preview = styled.button`
+  padding: 0;
+  margin: 0;
+  border: none;
+  background: transparent;
+  background: ${({ theme }) => theme.colors.bg.v7};
+  color: ${({ theme }) => theme.colors.fg.v1};
+`;
+
+const inputStyles = {
+  /* stylelint-disable-next-line rule-empty-line-before */
+  input: {
+    fontFamily: 'monospace',
+    width: '60px',
+  },
+};
+
+function EditableHexPreview({ hex, onChange }) {
+  const [isEditing, setIsEditing] = useState(false);
+
+  // Handle ESC keypress to toggle input field.
+  const handleKeyPress = useCallback(
+    (evt) => {
+      if ('Escape' === evt.key && isEditing) {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setIsEditing(false);
+      }
+    },
+    [isEditing]
+  );
+
+  useLayoutEffect(() => {
+    document.addEventListener('keydown', handleKeyPress);
```

Comment:
What happens here when the user clicks outside of this input? I don't see `onBlur` anywhere. Will this leak the global document key listener? We could merge `handleHeyPress` and the `useLayoutEffect`, condition it on `isEditing`, and bail out when `isEditing === false`. Then, at least, we will always get unsubscribe. But still, there should be a `onBlur` somewhere. Or is `onChangeComplete` handle `onBlur` internally?

Reply:
Yes, it still needs a blur handler.

Reply:
Makes sense, yes, it needs one. Since I cannot directly attach an `onBlur` to `EditableInput` (it's a div with an input in it), I might need to do some trickery using `document.activeElement` or just replace `EditableInput` with our own input.

Reply:
[`focusout`  bubbles](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusout_event).

Reply:
React doesn't support `focusout` though 😕 
See https://github.com/facebook/react/issues/6410

Reply:
Added a possible workaround in d8fcc3e8b58ecc98c0317ef926b15c02e970beb0